lib/puppet/type/user.rb

```diff
@@ -464,7 +468,9 @@ def insync?(current)
         }
       end
 
-      if obj = @parameters[:groups] and groups = obj.should
+      obj = @parameters[:groups]
+      groups = obj.should if obj
+      if groups
         autos += groups.split(",")
```

Comment:
Previously `split` was only called if both `obj` and `groups` were non-nil. Now it's only called if `groups` is non-nil. But `obj` can be `nil`, while `groups` could have been set above on line 449, and it will be an array, so calling [`Array#split` fails](https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/Acceptance%20Suites/view/master/view/Suite/job/platform_puppet-agent_puppet-agent-integration-suite_daily-master/637/RMM_COMPONENT_TO_TEST_NAME=puppet,SLAVE_LABEL=beaker,TEST_TARGET=osx1014-64a/console). I think line 473, needs to be `if obj and groups`.

Reply:
Submitted https://github.com/puppetlabs/puppet/pull/7597